x-pack/test/observability_functional/apps/observability/pages/cases/case_details.ts

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import expect from '@kbn/expect';
+import { CommentType } from '@kbn/cases-plugin/common/api';
+import { FtrProviderContext } from '../../../../ftr_provider_context';
+
+export default ({ getPageObjects, getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const cases = getService('cases');
+  const observability = getService('observability');
+  const find = getService('find');
+  const PageObjects = getPageObjects(['common', 'header']);
+
+  describe('Cases', () => {
+    before(async () => {
+      await esArchiver.load('x-pack/test/functional/es_archives/observability/alerts');
+      await esArchiver.load('x-pack/test/functional/es_archives/infra/metrics_and_logs');
+    });
+
+    after(async () => {
+      await esArchiver.unload('x-pack/test/functional/es_archives/infra/metrics_and_logs');
+      await esArchiver.unload('x-pack/test/functional/es_archives/observability/alerts');
+    });
+
+    describe('Case detail rule link', () => {
+      before(async () => {
+        await observability.users.setTestUserRole(
+          observability.users.defineBasicObservabilityRole({
+            observabilityCases: ['all'],
+            logs: ['all'],
+          })
+        );
+
+        const owner = 'observability';
+        const caseData = await cases.api.createCase({
+          title: 'Sample case',
+          owner,
+        });
+        await cases.api.createAttachment({
+          caseId: caseData.id,
+          params: {
+            alertId: ['alert-id'],
+            index: ['.internal.alerts-observability.alerts-default-000001'],
+            rule: { id: 'rule-id', name: 'My rule name' },
+            type: CommentType.alert,
+            owner,
+          },
+        });
+      });
+
+      after(async () => {
+        await cases.api.deleteAllCases();
+      });
+
+      it('should link to observability rule pages in case details', async () => {
+        await cases.navigation.navigateToApp('observabilityCases', 'cases-all-title');
+        await cases.casesTable.waitForCasesToBeListed();
+        await cases.casesTable.goToFirstListedCase();
+        await PageObjects.header.waitUntilLoadingHasFinished();
+
+        await (await find.byCssSelector('[data-test-subj*="alert-rule-link"]')).click();
+
+        expect(
+          await (await find.byCssSelector('[data-test-subj="breadcrumb first"]')).getVisibleText()
```

Comment:
@maryam-saeidi I added this part of code as a "hack" to get me unblocked in this [PR](https://github.com/elastic/kibana/pull/132479). If you have a look at this comment https://github.com/elastic/kibana/pull/132479#discussion_r876673905, best way to check if user landed in the o11y Rule details page is  `expect(await find.existsByCssSelector('[data-test-subj="ruleDetailPageTitle"]')).to.eql(true)`. This means that you will need to add a [data-test-subj](https://github.com/elastic/kibana/pull/132479/commits/82839526dd5326f118e6cad30879ae9dc7a09e0f#diff-d19c39560cbd9661c7d9df71eed8fc45a188ada9bc97e0b7c321da8f4afdfdd6R213) to the Rule Details page as well.

Could you try to use above code and see if it works?

Reply:
The issue is I didn't create the rule and related alert, so it will not load the title. Is it possible to validate the URL path? Is there a service for it?

Reply:
@maryam-saeidi Great! As discussed using what you currently have is much better!

```
const url = await browser.getCurrentUrl();
expect(url.includes('/app/observability/alerts/rules')).to.be(true);
```